### PR TITLE
perf(orgs): 默认组织改为实体组织，并支持全局组织

### PR DESCRIPTION
**Pull request by fit2bot**
perf(orgs): 默认组织改为实体组织

### CODE REVIEW

File: apps/orgs/api.py [hunk 1]
```diff
@@ -129,3 +132,16 @@ def perform_bulk_destroy(self, queryset):
...
+
+class CurrentOrgDetailApi(RetrieveAPIView):
+    serializer_class = CurrentOrgSerializer
+    permission_classes = (IsValidUser,)
+
+    def get_object(self):
+        return current_org
+
+    def get_permissions(self):
+        if not current_org.can_use_by(self.request.user):
```
Comment (xuxinwen) [inline apps/orgs/api.py:145]: 这里的权限可以在上面声明，逻辑与

File: apps/orgs/migrations/0010_auto_20210219_1241.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        ('xpack', [
+            'Account', 'SyncInstanceDetail', 'SyncInstanceTask', 'SyncInstanceTaskExecution',
+            'ChangeAuthPlan', 'ChangeAuthPlanExecution', 'ChangeAuthPlanTask',
+            'GatherUserTask', 'GatherUserTaskExecution',
+        ]),
+    ]
+    print("")
+    for app, models_name in org_app_models:
+        for model_name in models_name:
+            t_start = time.time()
```
Comment (xuxinwen) [inline apps/orgs/migrations/0010_auto_20210219_1241.py:39]: 这里显示的切换到 root 组织比较保险

File: apps/orgs/models.py [hunk 3]
```diff
@@ -179,37 +168,32 @@ def get_user_audit_orgs(cls, user):
...
+            return [cls.root(), *cls.objects.all()]
         return cls.get_user_orgs_by_role(user, ROLE.AUDITOR)
 
     @classmethod
     def get_user_admin_or_audit_orgs(cls, user):
         if user.is_anonymous:
             return cls.objects.none()
         if user.is_superuser or user.is_super_auditor:
-            return [*cls.objects.all(), cls.default()]
+            return [cls.root(), *cls.objects.all()]
```
Comment (xuxinwen) [inline apps/orgs/models.py:179]: `root` 组织要不要在这里

File: apps/orgs/mixins/models.py [hunk 4]
```diff
@@ -60,12 +57,10 @@ class OrgModelMixin(models.Model):
...
-        if org is None:
-            return super().save(*args, **kwargs)
-        if org.is_real() or org.is_system():
+        if org.is_root() and not self.org_id:
+            raise ValidationError('Please save in a organization')
+        if not org.is_root():
             self.org_id = org.id
-        elif org.is_default():
-            self.org_id = ''
         return super().save(*args, **kwargs)
```
Comment (BaiJiangJie) [inline apps/orgs/mixins/models.py:64]: check...

File: apps/orgs/api.py (hunk 1 above)
Comment (BaiJiangJie) [reply apps/orgs/api.py:145]: check...

File: apps/users/models/user.py [hunk 5]
```diff
@@ -168,7 +168,7 @@ def role_display(self):
     def org_roles(self):
         from orgs.models import ROLE as ORG_ROLE
 
-        if not current_org.is_real():
+        if not current_org.is_root():
```
Comment (BaiJiangJie) [inline apps/users/models/user.py:171]: check...

File: apps/assets/tasks/nodes_amount.py [hunk 6]
```diff
@@ -15,7 +15,7 @@
 @shared_task
 def check_node_assets_amount_task(orgid=None):
```
Comment (BaiJiangJie) [inline apps/assets/tasks/nodes_amount.py:16]: orgid 命名

File: apps/assets/models/node.py [hunk 7]
```diff
@@ -485,12 +486,20 @@ def create_org_root_node(cls):
         # 如果使用current_org 在set_current_org时会死循环
         ori_org = get_current_org()
         with transaction.atomic():
-            if not ori_org.is_real():
+            if ori_org.is_default():
```
Comment (BaiJiangJie) [inline apps/assets/models/node.py:489]: check...

File: apps/orgs/mixins/models.py (hunk 4 above)
Comment (BaiJiangJie) [reply apps/orgs/mixins/models.py:64]: org 可能是None

    def save(self, *args, **kwargs):
        org = get_current_org()
        if org is None:
            raise ValidationError('Please save in a organization')
        elif org.is_root():
            if not self.org_id:
                raise ValidationError('Please save in a organization')
        else:
            self.org_id = org.id
        return super().save(*args, **kwargs)

File: apps/users/models/user.py (hunk 5 above)
Comment (BaiJiangJie) [reply apps/users/models/user.py:171]: if current_org.is_root():
            if self.is_superuser:
                return [ORG_ROLE.ADMIN]
            elif self.is_super_auditor:
                return [ORG_ROLE.AUDITOR]
            else:
                return [ORG_ROLE.USER]

File: apps/assets/models/node.py [hunk 8]
```diff
@@ -485,12 +486,20 @@ def create_org_root_node(cls):
...
             return root
 
+    @classmethod
+    def org_root_nodes(cls):
+        nodes = cls.objects.filter(parent_key='') \
+            .filter(key__regex=r'^[0-9]+$') \
+            .exclude(key__startswith='-') \
+            .order_by('key')
+        return nodes
+
```
Comment (BaiJiangJie) [inline apps/assets/models/node.py:502]: @classmethod
    def create_org_root_node(cls):
        ori_org = get_current_org()
        with transaction.atomic():
            key = cls.get_next_org_root_node_key()
            root = cls.objects.create(key=key, value=ori_org.name)
            return root

    @classmethod
    def org_root(cls):
        org_roots = cls.org_root_nodes()
        if org_roots:
            return org_roots[0]
        ori_org = get_current_org()
        # 如果使用current_org 在set_current_org时会死循环
        if ori_org.is_root():
            root = cls.default_node()
        elif ori_org.is_default():
            root = cls.default_node()
        else:
            root = cls.create_org_root_node()
        return root

File: apps/orgs/api.py (hunk 1 above)
Comment (BaiJiangJie) [reply apps/orgs/api.py:145]: class UserCanUseCurrentOrg(permissions.BasePermission):
    def has_permission(self, request, view):
        return current_org.can_use_by(request.user)

**Review by BaiJiangJie — APPROVED** [review]

**Review by BaiJiangJie — APPROVED** [review]

**Pull request merged**